soroban-sdk/src/testutils.rs

```diff
@@ -413,15 +415,90 @@ pub mod budget {
     }
 }
 
+#[derive(Clone)]
+pub struct ContractEvent {
+    env: Env,
+    pub contract_id: crate::Address,
+    topics: Vec<Val>,
+    data: Val,
+}
+
+impl Eq for ContractEvent {}
+
+impl PartialEq for ContractEvent {
+    fn eq(&self, other: &Self) -> bool {
+        if self.contract_id != other.contract_id || self.topics != other.topics {
+            false
+        } else if self.data.is_object() || other.data.is_object() {
+            self.env.obj_cmp(self.data, other.data).unwrap_infallible() == 0
+        } else {
+            self.data.get_payload() == other.data.get_payload()
+        }
+    }
+}
+
+impl Debug for ContractEvent {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "ContractEvent {{ contract_id: {:?}, topics: {:?}, data: {:?} }}",
+            self.contract_id, self.topics, self.data
+        )
+    }
+}
+
+impl ContractEvent {
+    pub fn new(
```

Comment:
Is this internally created only? In the interest of keeping the public interface exposed tighter and just expose things as we need, can it be pub crate or doc hidden?